encoders/nlp/UniversalSentenceEncoder/__init__.py

```diff
@@ -17,21 +19,29 @@ class UniversalSentenceEncoder(BaseTFEncoder):
     def __init__(
             self,
             model_url: str = 'https://tfhub.dev/google/universal-sentence-encoder/4',
-            *args,
+            preprocessor_url: Optional[str] = None,
+            * args,
             **kwargs):
         """
         :param model_url: the url of the model (TensorFlow Hub). For supported models see
                           family overview: https://tfhub.dev/google/collections/universal-sentence-encoder/1)
+        :param preprocessor_url: the url of preprocessors (TensorFlow Hub).
         :param args:
         :param kwargs:
         """
         super().__init__(*args, **kwargs)
         self.model_url = model_url
+        self.preprocessor_url = preprocessor_url
 
     def post_init(self):
         self.to_device()
+        import tensorflow as tf
+        import tensorflow_text as text
         import tensorflow_hub as hub
-        self.model = hub.load(self.model_url)
+        self.preprocessor = None
```

Comment:
I see another problem...The workflow cmlm implies to select a result dictionary 'default'
 `english_embeds = encoder(preprocessor(english_sentences))["default"]`
but for the old code it is used 
`model.load()`
As far I know, It is necessary to split in different two use cases and it can not be reused how I pretend in the current PR. For me, it seems easier to create two different private functions for each case.

Reply:
and how would u detect each use case?

Reply:
In the end, you have two different pipelines that  depend on your input model (your type of encoder). My usual methodology to do this, it keeps all the time simple the code (KISS principle) and understand what it is defining these two use cases. As far I know, it is defined by a string path model.  Also, for having  the control of this path model, I would define them as a constant... you can  manage which  input variables are correct in the function...

Reply:
would it then make sense to add a totally brand new Executor? to habdle the different cases? I am afraid this will fall into having a lot of if ... else ...

Reply:
In the end, we are doing baby steps for refactoring... Yes, in the first iteration you will detect this technical debt with this one if / else....   but also , you have a general view of the pipeline... If you have a function with a if / else that its only responsability is tocall  two functions (no more), now, you understand that you class has two responsibilities / logic concepts (break the single responsability pattern), you need a `polymorphism` [pattern](https://stackoverflow.com/questions/47402390/polymorphism-vs-if-and-logic)... Which one? you can start to think with of your design patterns can fit better in your use case... 
Sorry if it is a big comment for a PR... Also... not forget all this redesign / refactor principles  are a fight between technical debt and clean code, you can keep it "dirty" waiting for the best moment to change it.

Reply:
Well the polymorphism exists from the BaseExecutor class all the way down to the hub executors. So here the only point is where these 2 logics will become so complex that the need for having another Executor is more clear.

You can try to do the `if ... else` logic and see how it looks.

As per the technical debt, it is not such a problem ih the `hub` as this is a `library` of executors specially thought to have a clear and small interface (`init`, `post_init`, `encode`) where one can fit its own logic. So as long as the usage and testing of this specific class is clear there should not be a problem

Reply:
Polymorphism could be applicated with a [composition](https://en.wikipedia.org/wiki/Composition_over_inheritance) structure also. It didn t review yet all your code, but it is a possibility.

Reply:
yes but we sacrifice polymorphism and code reuse to have no dependency between hub executors as much as possible so that they have no dependencies and can be seen as independent small projects

Reply:
I think you don t need to sacrifice this with a composition structure... but I admit I would need to check deeper the code to provide you a proposal...

Reply:
if it does not involve affecting other executors in the hub it would be okey